src/extensions/scratch3_text2speech/index.js

```diff
@@ -457,16 +627,29 @@ class Scratch3Text2SpeechBlocks {
     speakAndWait (args, util) {
         // Cast input to string
         let words = Cast.toString(args.WORDS);
-        let locale = this.localeToPolly(this.getCurrentLanguage());
+        let locale = this._getSpeechSynthLocale();
 
         const state = this._getState(util.target);
 
-        const gender = this.VOICE_INFO[state.voiceId].gender;
-        const playbackRate = this.VOICE_INFO[state.voiceId].playbackRate;
+        let gender = this.VOICE_INFO[state.voiceId].gender;
+        let playbackRate = this.VOICE_INFO[state.voiceId].playbackRate;
+
+        // Special case for voices where the synthesis service only provides a
+        // single gender voice. In that case, always request the female voice,
+        // and set special playback rates for the tenor and giant voices.
+        if (this.LANGUAGE_INFO[this.getCurrentLanguage()].singleGender) {
+            gender = 'female';
+            if (state.voiceId === TENOR_ID) {
+                playbackRate = FEMALE_TENOR_RATE;
+            }
+            if (state.voiceId === GIANT_ID) {
+                playbackRate = FEMALE_GIANT_RATE;
+            }
+        }
 
         if (state.voiceId === KITTEN_ID) {
             words = words.replace(/\S+/g, 'meow');
-            locale = 'en-US';
+            locale = this.LANGUAGE_INFO[this.DEFAULT_LANGUAGE].speechSynthLocale;
```

Comment:
Just confirming that I'm reading this right and the kitten voice is still always english.

Reply:
meow (yup)!